src/main/scala/org/hammerlab/guacamole/reads/PairedRead.scala

```diff
@@ -18,29 +18,11 @@ case class PairedRead[+T <: Read](read: T,
   override val token: Int = read.token
   override val failedVendorQualityChecks: Boolean = read.failedVendorQualityChecks
   override val sampleName: String = read.sampleName
-  override val isPositiveStrand: Boolean = read.isPositiveStrand
   override val baseQualities: Seq[Byte] = read.baseQualities
   override val isDuplicate: Boolean = read.isDuplicate
   override val sequence: Seq[Byte] = read.sequence
   override val isPaired: Boolean = true
   override val isMapped = read.isMapped
+  override def asMappedRead = read.asMappedRead
   override val hasMdTag = read.hasMdTag
-}
-
-object PairedRead {
-  def apply(record: SAMRecord,
-            token: Int,
-            requireMDTagsOnMappedReads: Boolean,
-            referenceGenome: Option[ReferenceGenome]): PairedRead[Read] = {
-    val read = Read.fromSAMRecord(
-      record,
-      token,
-      requireMDTagsOnMappedReads,
-      referenceGenome
-    )
-
-    val mateAlignment = MateAlignmentProperties(record)
-    PairedRead(read, isFirstInPair = record.getFirstOfPairFlag, mateAlignment)
-  }
-}
-
```

Comment:
Yeah, I'm thinking of the `MappedRead`, `PairedRead`, `UnmappedRead`, and `PairedMappedRead` classes (the latter is only used I think in the SV caller, and is kind of different from the others, possibly needs to be reconsidered if necessary). Currently all of these classes except `PairedRead` have no BAM/SAM/ADAM-specific code in them; this change moves that code out of `PairedRead`  to make it consistent with the others. I agree that `Read.scala` is a monster and we may want to refactor that, perhaps to split the BAM/SAM/ADAM low-level stuff from the high-level `Read` trait, but I think that's better done in a separate PR.